packages/flutter/test/material/switch_theme_test.dart

```diff
@@ -0,0 +1,288 @@
+    const MaterialTapTargetSize materialTapTargetSize = MaterialTapTargetSize.shrinkWrap;
+    const Color focusColor = Color(0xfffffff4);
+    const Color hoverColor = Color(0xfffffff5);
+    const double splashRadius = 1.0;
+
+    Widget buildSwitch({bool selected = false, bool autofocus = false}) {
+      return MaterialApp(
+        theme: ThemeData(
+          switchTheme: SwitchThemeData(
+            thumbColor: MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+              if (states.contains(MaterialState.selected)) {
+                return selectedThumbColor;
+              }
+              return defaultThumbColor;
+            }),
+            trackColor: MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+              if (states.contains(MaterialState.selected)) {
+                return selectedTrackColor;
+              }
+              return defaultTrackColor;
+            }),
+            mouseCursor: mouseCursor,
+            materialTapTargetSize: materialTapTargetSize,
+            focusColor: focusColor,
+            hoverColor: hoverColor,
+            splashRadius: splashRadius,
+          ),
+        ),
+        home: Scaffold(
+          body: Switch(
+            dragStartBehavior: DragStartBehavior.down,
+            value: selected,
+            onChanged: (bool value) {},
+            autofocus: autofocus,
+          ),
+        ),
+      );
+    }
+
+    // Switch.
+    await tester.pumpWidget(buildSwitch());
+    await tester.pumpAndSettle();
+    expect(
+      _getSwitchMaterial(tester),
+      paints
+        ..rrect(color: defaultTrackColor)
+        ..circle()
+        ..circle()
+        ..circle()
+        ..circle(color: defaultThumbColor),
+    );
+    // Size from MaterialTapTargetSize.shrinkWrap.
+    expect(tester.getSize(find.byType(Switch)), const Size(59.0, 40.0));
+
+    // Selected switch.
+    await tester.pumpWidget(buildSwitch(selected: true));
+    await tester.pumpAndSettle();
+    expect(
+      _getSwitchMaterial(tester),
+      paints
+        ..rrect(color: selectedTrackColor)
+        ..circle()
+        ..circle()
+        ..circle()
+        ..circle(color: selectedThumbColor),
+    );
+
+    // Switch with hover.
+    await tester.pumpWidget(buildSwitch());
+    await _pointGestureToSwitch(tester);
+    await tester.pumpAndSettle();
+    expect(RendererBinding.instance!.mouseTracker.debugDeviceActiveCursor(1), SystemMouseCursors.text);
+    expect(_getSwitchMaterial(tester), paints..circle(color: hoverColor));
+
+    // Switch with focus.
+    await tester.pumpWidget(buildSwitch(autofocus: true));
+    await tester.pumpAndSettle();
+    expect(_getSwitchMaterial(tester), paints..circle(color: focusColor, radius: splashRadius));
+  });
+
+  testWidgets('Switch properties are taken over the theme values', (WidgetTester tester) async {
+    tester.binding.focusManager.highlightStrategy = FocusHighlightStrategy.alwaysTraditional;
```

Comment:
Do yuo have to un-set this at the end of the test? here and all other tests that change this in this PR

Reply:
I searched in the code base and it does not seem to be done in any tests. It seems to reset to the default after each widget test if I try to remove it.